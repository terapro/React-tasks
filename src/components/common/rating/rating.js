import React from 'react';
import PropTypes from 'prop-types';


const Rating = ({value}) => {

  return (
    <span className='rating'>
      {value}
    </span>
  )

};

Rating.propTypes = {
  value: PropTypes.number
};
Rating.defaultProps = {
  value: 0
};

export {Rating};
