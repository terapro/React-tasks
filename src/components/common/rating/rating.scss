.rating {
  display: inline-block;
  color: aquamarine;
  border: 1px solid aquamarine;
  border-radius: 100%;
  font-size: 18px;
  margin-left: $elementPadding;
  height: $elementHeight + $elementPadding;
  width: $elementHeight + $elementPadding;
  line-height: $elementHeight +$elementPadding;
  text-align: center;
}