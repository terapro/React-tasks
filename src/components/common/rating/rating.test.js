import React from 'react';
import {Rating} from 'src/components/common/rating/rating.js';
import {shallow} from 'enzyme';

const mockNumber = '5';

describe('<Rating />', () => {
  it('should render the logo', () => {
    const wrapper = shallow(<Rating value={mockNumber}/>);
    expect(wrapper).toMatchSnapshot();
  });
});