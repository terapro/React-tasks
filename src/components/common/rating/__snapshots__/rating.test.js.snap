// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should render the logo 1`] = `
<span
  className="rating"
>
  5
</span>
`;
